src/data/markdown/docs/02 javascript api/11 jslib/functional/10 error handling.md

```diff
@@ -0,0 +1,73 @@
+---
+title: 'Error handling.'
+description: 'How to handle errors in functional.js.'
+---
+
+When executing a performance or integration test, you should expect that your system under test may crash. If this happens, your test should print useful information rather than stack traces caused by unexpected HTTP responses. 
+
+`functional` library is designed to make it easy to write test scripts that are resilient to failing SUT (System under test). 
+
+It's not uncommon for performance testers to write fragile code that assumes the http response will contain expected data. 
+
+Fragile code is most clearly demonstrated with an example. 
+
+<CodeGroup labels={["Test code that is fragile to failing SUT"]}>
+
+```javascript
+import { check, group } from 'k6';
+import http from 'k6/http';
+
+export default function() {
+  group("Fetch a list of public crocodiles", () => {
+    let res = http.get("https://test-api.k6.io/public/crocodiles");
+
+    check(res, {
+      'is status 200': (r) => r.status === 200,
+      'got more than 5 crocs': (r) => r.json().length > 5,
+    });
+  })
+  // more code here
+}
+```
+
+</CodeGroup>
+
+
+This code will work fine as long as SUT (System Under Test) returns correct responses. When the SUT starts to fail, there's a good chance the `r.json().length` will throw an exception similar to 
+
+```bash
+ERRO[0001] GoError: cannot parse json due to an error at line 1, character 2 , error: invalid character '<' looking for beginning of value
+running at reflect.methodValueCall (native)
+default at gotMoreThan5Crocs (file:///home/user/happy-path-check.js:7:68(5))
+  at github.com/loadimpact/k6/js/common.Bind.func1 (native)
+  at file:///home/user/happy-path-check.js:5:22(17)  executor=per-vu-iterations scenario=default source=stacktrace
+```
+
+In this example, the system was overloaded, and the load balancer returned a 503 response that did not have a valid JSON body. k6 has thrown a JavaScript exception and restarted execution from the beginning. 
+This test code is fragile to failing SUT because the first `check` does not prevent the second check from executing. 
+It's possible to rewrite this code to be less fragile, but that will make it longer and less readable. 
+
+Error handling of this type happens automatically when using the `functional.js` library.
+When the first `expect` fails, the remaining checks in the chain are not executed, and the test is marked as failed — the execution proceeds to the next `test()` instead of restarting from the top.
+
```

Comment:
I think the naming `test` is super confusing ... so I am with @simskij on https://github.com/k6io/docs/pull/230#discussion_r589320904

Reply:
Agreed. I'll rename it to something less ambiguous.